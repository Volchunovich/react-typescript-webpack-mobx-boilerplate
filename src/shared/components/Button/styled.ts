const PrimaryStyled = `
  // Primary styles

  background: linear-gradient(to bottom,#9239d1 50%,#8239d1 50%);
  border-bottom: 8px solid #6b278e
`;

const LightStyled = `
  // Light styles

  background: linear-gradient(to bottom,#a4d4ff 50%,#91c6f5 50%);
  border-bottom: 8px solid #4e0a71;
`;

const DarkStyled = `
  // Dark styles

  background: linear-gradient(to bottom, #d1394e 50%, #cd2f46 50%);
  border-bottom: 8px solid #832154;
  
  &:hover {
    background: linear-gradient(to bottom,#5c40ab 50%,#4d3690 50%);
    border-bottom: 8px solid #290f94;
    opacity: .9;
    color: #ff0000;
    font-weight: 500;
    font-size: 100px;
    padding: 0 0 30px 0;
    height: 95px;
    line-height: 0;
  }
`;

const Styles = {
  PrimaryStyled,
  LightStyled,
  DarkStyled,
};

export default Styles;
