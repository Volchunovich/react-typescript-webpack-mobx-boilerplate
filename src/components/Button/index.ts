import styled from 'styled-components';
import { ButtonStyledProps } from './utils/button.types';
import getTheme from './utils/helpers.utils';

const Button = styled.button<ButtonStyledProps>`
  ${(props: ButtonStyledProps) => getTheme(props)};

  min-width: 350px;
  margin: 0 20px;
  text-align: center;
  font-size: 60px;
  font-family: 'Courier New', sans-serif;
  height: 90px;
  line-height: 90px;
  color: #fff;

  &:hover {
    cursor: pointer;
  }
  &:focus {
    outline: none;
  }
`;

export default Button;
