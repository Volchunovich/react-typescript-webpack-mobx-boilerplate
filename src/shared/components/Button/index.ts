import styled from 'styled-components';
import getTheme from './utils/helpers.utils';

export interface ButtonStyledProps {
  primary?: boolean;
  light?: boolean;
}

const Button = styled.button<ButtonStyledProps>`
  ${(props: ButtonStyledProps) => getTheme(props)};

  min-width: 350px;
  margin: 0 20px;
  text-align: center;
  font-size: 60px;
  font-family: 'ATCKrueger', sans-serif;
  height: 90px;
  line-height: 90px;
  color: #fff;

  &:hover {
    cursor: pointer;
  }
`;

export default Button;
