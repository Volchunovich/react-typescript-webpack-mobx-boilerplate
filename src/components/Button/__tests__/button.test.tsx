import { shallow } from 'enzyme';
import * as React from 'react';
import Button from '../index';

describe('Button', () => {
  describe('StyledComponent', () => {
    it('Should render without errors', () => {
      const component = shallow(<Button>Without errors</Button>).instance();
      expect(component).toMatchSnapshot();
    });

    it('Primary prop should be true', () => {
      const component = shallow(<Button buttonType='primary'>Primary</Button>);
      const prop = component.prop('buttonType');
      expect(prop).toEqual('primary');
      expect(prop).toMatchSnapshot();
    });

    it('Light prop should be true', () => {
      const component = shallow(<Button buttonType='light'>Light</Button>);
      const prop = component.prop('buttonType');
      expect(prop).toEqual('light');
      expect(prop).toMatchSnapshot();
    });
  });
});
