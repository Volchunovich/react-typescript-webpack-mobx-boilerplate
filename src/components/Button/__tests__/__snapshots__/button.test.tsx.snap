// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button StyledComponent Light prop should be true 1`] = `"light"`;

exports[`Button StyledComponent Primary prop should be true 1`] = `"primary"`;

exports[`Button StyledComponent Should render without errors 1`] = `null`;
